gramps/gen/filters/optimizer.py

```diff
@@ -43,95 +44,73 @@ def union(sets):
         return set()
 
 
+def symmetric_difference(sets):
```

Comment:
You could write this function as:
```
def symmetric_difference(sets):
    return reduce(lambda x, y: x^y, sets, set())
```

This will need `from functools import reduce` added to the imports.

The same approach could be used for _union_ and _intersection_.

Reply:
Nice suggestion @Nick-Hall!
I've implemented this. For intersection I've avoided computing x&x when we only have a single set.